core/src/main/scala/cats/Foldable.scala

```diff
@@ -372,4 +378,12 @@ object Foldable {
       Eval.defer(if (it.hasNext) f(it.next, loop()) else lb)
     loop()
   }
+
+  def iteratorFoldM[M[_], A, B](it: Iterator[A], z: B)(f: (B, A) => M[B])(implicit M: Monad[M]): M[B] = {
```

Comment:
How about we also mention here in the doc the incentive of this special implementation for `Iteratable`s?